test/controllers/collection_numbers_controller_test.rb

```diff
@@ -210,14 +206,14 @@ def test_create_collection_number_redirect
     # Prove that query params are added to form action.
     login(obs.user.login)
     get(:new, params)
-    assert_select("form[action*='number/#{obs.id}?q=#{q}']")
+    assert_select("form[action*='numbers/#{obs.id}?q=#{q}']")
 
     # Prove that post keeps query params intact.
-    post(:new, params)
-    assert_redirected_to(obs.show_link_args.merge(q: q))
+    post(:create, params)
+    assert_redirected_to("#{observations_path}/#{obs.id}?q=#{q}")
```

Comment:
Is there a more idiomatic way to do this?

Reply:
This should work:
`assert_redirected_to observation_path(obs.id, q: q)`
[Api docs reference](https://api.rubyonrails.org/classes/ActionDispatch/Assertions/ResponseAssertions.html)

Reply:
Note that the path name is singular for a :show link, plural for :index.
`:show -> observation_path(:id), :index -> observations_path`

Reply:
This is a good place to use other path syntax, too. You can write (in line 212)
`post new_collection_number_path(params)`
The `new_*` paths can distinguish between the `get` and `post` verbs - 
`get new_collection_number_path` renders the`:new` form, `post new_collection_number_path` `:create`s the collection number, via the same path.
Likewise, for `edit_*` paths,
`get` renders the `:edit` form, `post` `:update`s the collection number, via the same path

Reply:
I tried this, but am not getting the syntax write.
Because things aren't actually broken in the current syntax, I'm going to move on and leave this refactoring for another day/PR.